c/kastore.c

```diff
@@ -908,6 +908,12 @@ kastore_bput(kastore_t *self, const char *key, size_t key_len, const void *array
     if (ret != 0) {
         goto out;
     }
+    if (array == NULL) {
```

Comment:
Is this necessary? Isn't the problem just that the later assert is wrong (or at least should be checking that ``array_len == 0`` when both are NULL?

Reply:
Changed my mind in meeting, let's keep it this way for symmetry with the other code path.